src/index.ts

```diff
@@ -156,6 +158,24 @@ export default class Matic extends ContractsBase {
     return this.depositManager.depositEther(amount, options)
   }
 
+  async depositDataByHash(txHash: string) {
+    const depositReceipt = await this.web3Client.parentWeb3.eth.getTransactionReceipt(txHash)
+    if(!depositReceipt) {
+      return 'Transaction hash is not Found'
+    }
+    const newDepositEvent = depositReceipt.logs.find(
```

Comment:
We have a bulk deposit functionality as well, so there can be multiple deposit events in a single receipt.
You'll need to `find` on both `NEW_DEPOSIT_EVENT_SIG` and deposit id.

Reply:
I don't have`depositId` here
I am doing all this to get `depositId` from `transactionHash`
any other way to do it? @atvanguard

Reply:
Then you'd need to return an array of depositIds. Sorry should have pointed out earlier.